tests/virt_autotest/update_package.pm

```diff
@@ -53,8 +53,10 @@ sub run {
         set_grub_on_vh('', '', 'xen') if is_xen_host;
         set_grub_on_vh('', '', 'kvm') if is_kvm_host;
     } else {
-        my @files_to_upload = ("/boot/grub2/grub.cfg", "/etc/default/grub");
-        upload_logs($_, failok => 1) foreach (@files_to_upload);
+        if (!is_s390x) {
+            my @files_to_upload = ("/boot/grub2/grub.cfg", "/etc/default/grub");
+            upload_logs($_, failok => 1) foreach (@files_to_upload);
+        }
     }
```

Comment:
I am curious about the change here. Why it enters into this branch for s390x with 15-SP6 ?  15-SP5 testing did not enter into this branch, so test can pass, see https://openqa.suse.de/tests/11201381#step/update_package/17.

Reply:
Me neither.  Refer to current test failure - https://openqa.suse.de/tests/13084593#step/update_package/6

```
# Test died: command 'curl --form upload=@/boot/grub2/grub.cfg --form upname=update_package-grub.cfg http://worker34.oqa.prg2.suse.org:20113/M4ZLCQwGVfjlPKaC/uploadlog/grub.cfg' timed out at /usr/lib/os-autoinst/testapi.pm line 985.
	testapi::script_run("curl --form upload=\@/boot/grub2/grub.cfg --form upname=update"..., 90) called at /usr/lib/os-autoinst/testapi.pm line 2155
	testapi::upload_logs("/boot/grub2/grub.cfg", "failok", 1) called at sle/tests/virt_autotest/update_package.pm line 57
	update_package::run(update_package=HASH(0x55d589a4cd30)) called at /usr/lib/os-autoinst/basetest.pm line 352
	eval {...} called at /usr/lib/os-autoinst/basetest.pm line 346
	basetest::runtest(update_package=HASH(0x55d589a4cd30)) called at /usr/lib/os-autoinst/autotest.pm line 415
	eval {...} called at /usr/lib/os-autoinst/autotest.pm line 415
	autotest::runalltests() called at /usr/lib/os-autoinst/autotest.pm line 272
	eval {...} called at /usr/lib/os-autoinst/autotest.pm line 272
	autotest::run_all() called at /usr/lib/os-autoinst/autotest.pm line 323
	autotest::__ANON__(Mojo::IOLoop::ReadWriteProcess=HASH(0x55d58a2b7068)) called at /usr/lib/perl5/vendor_perl/5.26.1/Mojo/IOLoop/ReadWriteProcess.pm line 329
	eval {...} called at /usr/lib/perl5/vendor_perl/5.26.1/Mojo/IOLoop/ReadWriteProcess.pm line 329
	Mojo::IOLoop::ReadWriteProcess::_fork(Mojo::IOLoop::ReadWriteProcess=HASH(0x55d58a2b7068), CODE(0x55d58a103180)) called at /usr/lib/perl5/vendor_perl/5.26.1/Mojo/IOLoop/ReadWriteProcess.pm line 492
	Mojo::IOLoop::ReadWriteProcess::start(Mojo::IOLoop::ReadWriteProcess=HASH(0x55d58a2b7068)) called at /usr/lib/os-autoinst/autotest.pm line 325
	autotest::start_process() called at /usr/lib/os-autoinst/OpenQA/Isotovideo/Runner.pm line 94
	OpenQA::Isotovideo::Runner::start_autotest(OpenQA::Isotovideo::Runner=HASH(0x55d58348f760)) called at /usr/bin/isotovideo line 133
	eval {...} called at /usr/bin/isotovideo line 122
```

s390x test does enter intio this  `else` branch

Reply:
s390x test should not enter into else branch, you need to modify unless branch to make s390x test enters into it. Just like before, 15-SP5 test run.

Reply:
It seems that 15-SP5 test does not enter into either unless or else branch. The whole unless/else branch changed afterwards. So no need to modify unless branch either. Maybe a multiple branches if/unless block is better ?  For example:

```
if
elsif
else
```

Reply:
Sure, good point.

Reply:
@waynechen55 refer to commit - https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/8d17d7c0f30e5ed724d2169fec170328a652ce85, s390 need to use `upload_logs()` to upload "/boot/grub2/grub.cfg", "/etc/default/grub". However, s390x lpar does not support `upload_logs()` func.

Reply:
Use
```
if
elsif
else
```
to skip upload_logs for s390x test. You can have better solution if any. I just do not prefer to embed another if...else block inside current unless. Could you help finish this pull request ASAP ? It would be better this can be merged before next milestone test. @guoxuguang

Reply:
Sure, fix up VLAN network problem on s390x test environment and continue to finish up this PR asap.